src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -2,23 +2,109 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 
 namespace Microsoft.Extensions.Logging.Console
 {
     public class ConsoleLoggerProvider : ILoggerProvider
     {
+        private readonly ConcurrentDictionary<string, ConsoleLogger> _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
+
         private readonly Func<string, LogLevel, bool> _filter;
-        private readonly bool _includeScopes;
+        private readonly IConsoleLoggerSettings _settings;
 
         public ConsoleLoggerProvider(Func<string, LogLevel, bool> filter, bool includeScopes)
         {
+            if (filter == null)
+            {
+                throw new ArgumentNullException(nameof(filter));
+            }
+
             _filter = filter;
-            _includeScopes = includeScopes;
+            _settings = new ConsoleLoggerSettings()
+            {
+                IncludeScopes = includeScopes,
+            };
+        }
+
+        public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            _settings = settings;
+
+            if (_settings.ChangeToken != null)
+            {
+                _settings.ChangeToken.RegisterChangeCallback(OnConfigurationReload, null);
+            }
+        }
+
+        private void OnConfigurationReload(object state)
+        {
+            _settings.Reload();
```

Comment:
do we need to reload the values explicitly here...like if we there was a event(or change token) which said like `OnSettingsChanged`, we should be having a reloaded/changed settings already and we could just use them...
For example: https://github.com/aspnet/Configuration/pull/320/files#diff-3716cadb2612035150f3750b7e26ba58R84, we are already reloading them here and the `ConfigurationConsoleLoggerSettings` is just falling back on these changed settings...yeah, I know that settings could be built without configuration too, but thought we could make the `IConsoleLoggerSettings` interface a bit simpler...